iOS/rainbow/Controller/Camera/CameraController.swift

```diff
@@ -22,28 +58,259 @@ class CameraController: LuminaViewController {
+//        guard let cachedScoreEntry = cachedScoreEntry else {
+//            return false
+//        }
+//        if cachedScoreEntry.startDate != nil {
+//            return cachedScoreEntry.finishDate == nil
+//        } else {
+//            return false
+//        }
+    }
+    
+    fileprivate func hideStartView(passedView: UIView) {
+        UIView.animate(withDuration: 0.2, animations: {
+            passedView.alpha = 0.0
+        }, completion: { _ in
+            passedView.removeFromSuperview()
+            self.startCamera()
+        })
+    }
+    
+    fileprivate func showStartView() {
+        let startView = GameStartView(frame: self.view.frame)
+        startView.delegate = self
+        startView.gameStartButton.backgroundColor = UIColor.RainbowColors.orange
+        view.addSubview(startView)
+    }
+}
+
+protocol GameStartViewDelegate: class {
+    func gameStartViewButtonTapped(passedView: UIView)
+}
+
+extension CameraController: GameStartViewDelegate {
+    func gameStartViewButtonTapped(passedView: UIView) {
+        hideStartView(passedView: view)
     }
 }
 
 extension CameraController: LuminaDelegate {
     func streamed(videoFrame: UIImage, with predictions: [LuminaRecognitionResult]?, from controller: LuminaViewController) {
+        if gameState == .objectDetected {
+            return
+        }
         guard let bestName = predictions?.first?.predictions?.first?.name else {
             return
         }
         guard let bestConfidence = predictions?.first?.predictions?.first?.confidence else {
             return
         }
         if bestConfidence >= 0.9 {
-            self.textPrompt = "Detected: \(bestName) (\(bestConfidence))"
+            var objects = [ObjectEntry]()
+            if let cachedObjects = cachedScoreEntry?.objects {
+                objects = cachedObjects
+            }
+            let filteredObjects = objects.filter { $0.name == bestName }
+            if filteredObjects.count > 0 {
+                continueScanning()
+                return
+            }
+            gameState = .detectionInProgress
+            self.textPrompt = "Detecting: \(bestName)"
+            consecutiveDetectionCount += 1
+            if consecutiveDetectionCount > 4 {
+                objectDetected(label: bestName)
+            }
         } else {
-            self.textPrompt = ""
+            continueScanning()
+        }
+    }
+    
+    private func continueScanning() {
+        self.consecutiveDetectionCount = 0
+        gameState = .nothingDetected
+    }
+}
+
+extension CameraController {
+    func objectDetected(label: String) {
+        self.consecutiveDetectionCount = 0
+        gameState = .objectDetected
+        updateObjectUI(for: label)
+        AudioServicesPlayAlertSound(SystemSoundID(kSystemSoundID_Vibrate))
+    }
+    
+    func updateObjectUI(for label: String) {
+        textPrompt = "You found the \(label)!"
+        switch label {
+        case "Apple":
```

Comment:
this is where we'd have to loop through the array and find the corresponding image.